servers/apps/test/datetime_test.c

```diff
@@ -0,0 +1,41 @@
+#include <resea/malloc.h>
+#include <resea/printf.h>
+#include <resea/datetime.h>
+#include <string.h>
+#include "test.h"
+
+#define TEST_START_TIME 1577836800 // January 1 2020 12:00:00 AM
+#define TEST_END_TIME   1577923200 // January 2 2020 12:00:00 AM
+
+void datetime_test(void) {
+    uint64_t timestamp = TEST_START_TIME;
+    struct datetime datetime;
+    timestamp_to_datetime(timestamp, &datetime);
+
+    while (timestamp <= TEST_END_TIME) {
+        timestamp_to_datetime(timestamp, &datetime);
+        TEST_ASSERT(datetime.second >= 0 && datetime.second < 60);
+        TEST_ASSERT(datetime.minute >= 0 && datetime.minute < 60);
+        TEST_ASSERT(datetime.hour >= 0 && datetime.hour < 24);
+        TEST_ASSERT(datetime.month == 1);
+        TEST_ASSERT(datetime.year == 2020);
+        ++timestamp;
+    }
```

Comment:
This loop runs 86400 times and I think we don't need to do that. Instead, I recommend add some tests without a loop:

- normal case: converting to/from a common case, say, `2020-11-20T09:16:55`.
- edge / boundary cases: your *"Test for start of epoch time"* is a good example.
- corner case: e.g. Feb 29 in a leap year.

Reply:
I modified the tests as you requested. Please have a look again.

Reply:
Looks good! Thanks!